### DISCUSSION TITLE
Stop Keycloak from running event expiration queries

### DISCUSSION THREAD

**Discussion by vladgardusseedblink** [keycloak/keycloak#24786] [Q&A] [answered]
Hello,

I am using keycloak hosted on AWS, using a RDS aurora serverless cluster. What I want on the lower environments (dev, demo, etc.) is for those aurora clusters to scale to 0 so that we don’t pay the costs while there are no users that actually use keycloak for that environment.

The issue is that once every 5 or 10 minutes, I see in the debug logs that some queries start running at some give interval. Example of query: “select distinct realm.eventsExpiration from org.keycloak.models.jpa.entities.RealmEntity realm where realm.eventsExpiration > 0”

Because keycloak runs those queries periodically, the aurora serverless cluster is prevented from scaling to 0, so we end up paying for 24h hours per day of uptime.

Is there any way to increase the interval for which those queries run? Or can we disable it at all?

We are using this docker image: [Quay](http://quay.io/keycloak/keycloak:21.1.1). I also tried disabling health check, metrics, setting all Sessions settings in all realms to 30 days and all token settings for all realms to 30 days.

**Comment by danielFesenmeyer** [comment, marked as answer]
I think I've answered that here: https://github.com/keycloak/keycloak/discussions/21864#discussioncomment-6591895
Did not get feedback, but maybe you can check if it helps.

**Reply by vladgardusseedblink** [reply]
Thanks Daniel for the quick reply. Had a look over this thread, but I did not enable user/admin events in any of the realms. Does the schedule still run if all events are disabled?

**Reply by danielFesenmeyer** [reply]
Yes, it runs even when none of the realms has events enabled. And I think it makes sense for the case when events were enabled, were disabled in the meantime, and there are still old events stored.

**Reply by danielFesenmeyer** [reply]
Also note that there's currently just one setting for all of the cleanups here: https://github.com/keycloak/keycloak/blob/8da7a52969f5dfa1235227646ffb07d6fcee8ef3/model/legacy-private/src/main/java/org/keycloak/storage/datastore/LegacyDatastoreProviderFactory.java#L105

It means when increasing the interval, ClearExpiredClientInitialAccessTokens and ClearExpiredUserSessions will also run less frequently.

**Reply by vladgardusseedblink** [reply]
Yeah, makes sense. I will try out what you proposed in the other thread and mark as the answer. I only plan to make those changes on lower environments, where those clear schedules are not that important.

**Reply by vladgardusseedblink** [reply]
I don't see any env variable related to the schedule interval in the provider configuration from this link: https://www.keycloak.org/server/all-provider-config . Any I missing something?

**Reply by vladgardusseedblink** [reply]
I added `KC_SPI_SCHEDULED_INTERVAL` to my env variables and it worked. Going to mark this as the answer. Thanks Daniel for your help!
